File: ProblemSolver/ProblemSolver/src/domains/puzzle/PuzzleMover.java
package domains.puzzle;

import domains.dummy.DummyState;
import framework.problem.Mover;
import framework.problem.State;

/**
 *
 * @author Mike
 */
public class PuzzleMover extends Mover{
        public static final String SLIDE1 = "SLIDE TILE 1";
        public static final String SLIDE2 = "SLIDE TILE 2";
        public static final String SLIDE3 = "SLIDE TILE 3";
        public static final String SLIDE4 = "SLIDE TILE 4";
        public static final String SLIDE5 = "SLIDE TILE 5";
        public static final String SLIDE6 = "SLIDE TILE 6";
        public static final String SLIDE7 = "SLIDE TILE 7";
        public static final String SLIDE8 = "SLIDE TILE 8";
        
        

        public PuzzleMover() {
            super.addMove(SLIDE1, s -> trySlide1(s));
            super.addMove(SLIDE2, s -> trySlide2(s));
            super.addMove(SLIDE3, s -> trySlide3(s));
            super.addMove(SLIDE4, s -> trySlide4(s));
            super.addMove(SLIDE5, s -> trySlide5(s));
            super.addMove(SLIDE6, s -> trySlide6(s));
            super.addMove(SLIDE7, s -> trySlide7(s));
            super.addMove(SLIDE8, s -> trySlide8(s));
            
            
        }

        private State trySlide1(State state) {
            return null;
        }
        
        private State trySlide2(State state) {
            return null;
        }
        
        private State trySlide3(State state) {
            return null;
        }
        
        private State trySlide4(State state) {
            return null;
        }
        
        private State trySlide5(State state) {
            return null;
        }
        
        private State trySlide6(State state) {
            return null;
        }
        
        private State trySlide7(State state) {
            return null;
        }

        private State trySlide8(State state) {
            return null;
        }

        private State illegalMove(State state) {
            return null;
        }
}

File: Labs/lab 11 - Javascript/JSProblemSolver/js/domains/puzzle/PuzzleMover.js
const SLIDE_1 = "Slide 1";
const SLIDE_2 = "Slide 2";
const SLIDE_3 = "Slide 3";
const SLIDE_4 = "Slide 4";
const SLIDE_5 = "Slide 5";
const SLIDE_6 = "Slide 6";
const SLIDE_7 = "Slide 7";
const SLIDE_8 = "Slide 8";

var puzzleMover = new Mover();

puzzleMover.addMove(SLIDE_1,(s) => slideOne(s)); // You provide move functions
puzzleMover.addMove(SLIDE_2,(s) => slideTwo(s));
puzzleMover.addMove(SLIDE_3,(s) => slideThree(s));
puzzleMover.addMove(SLIDE_4,(s) => slideFour(s));
puzzleMover.addMove(SLIDE_5,(s) => slideFive(s));
puzzleMover.addMove(SLIDE_6,(s) => slideSix(s));
puzzleMover.addMove(SLIDE_7,(s) => slideSeven(s));
puzzleMover.addMove(SLIDE_8,(s) => slideEight(s));

function PuzzleMover() { }

PuzzleMover.prototype = puzzleMover;

function slideOne(state) {  // creates a new state with block moved
    
    return state;
}

function slideTwo(state) {  // creates a new state with block moved
    
    return state;
}

function slideThree(state) {  // creates a new state with block moved
    
    return state;
}

function slideFour(state) {  // creates a new state with block moved
    
    return state;
}

function slideFive(state) {  // creates a new state with block moved
    
    return state;
}

function slideSix(state) {  // creates a new state with block moved
    
    return state;
}

function slideSeven(state) {  // creates a new state with block moved
    
    return state;
}

function slideEight(state) {  // creates a new state with block moved
    
    return state;
}